simulation/test_http_connection.cpp

```diff
@@ -598,6 +598,55 @@ TORRENT_TEST(http_connection_http_error)
 	test_proxy_failure(settings_pack::http);
 }
 
+// Requests a proxied SSL connection. This test just ensures that the correct CONNECT request
+// is sent to the proxy server.
+TORRENT_TEST(http_connection_ssl_proxy)
+{
+	using sim::asio::ip::address_v4;
+	sim_config network_cfg;
+	sim::simulation sim{network_cfg};
+
+	sim::asio::io_service client_ios(sim, address_v4::from_string("10.0.0.1"));
+	sim::asio::io_service server_ios(sim, address_v4::from_string("10.0.0.2"));
```

Comment:
one way to simplify this test would be to remove the server and make the proxy just a http handler that verifies it gets the appropriate request to proxy. i.e. that the address is correct. Now you're also testing that the test-proxy implementation is working (which I don't think is strictly necessary to test for this)